src/components/voting/VoterButtons.tsx

```diff
@@ -99,55 +110,58 @@ const VoterButton = ({
       : changeReactionTx
     }
     params={buildTxParams()}
-    onClick={() => updateOrDelete(isDelete)}
+    onClick={() => {
+      updateOrDelete(isActive)
+      upserPost(getPostStructWithUpdatedCounts({ post, oldReaction, newKind }))
+    }}
     onSuccess={(txResult) => {
       reloadPost(args)
-
+      
       const newReactionId = reactionId || getNewIdsFromEvent(txResult)[1]?.toString()
-      updateOrDelete(isDelete, newReactionId)
+      updateOrDelete(isActive, newReactionId)
       onSuccess && onSuccess()
     }}
     onFailed={() => {
-      upsertReaction({ id, ...reaction })
+      oldReaction && upsertReaction(oldReaction)
+      upserPost(post)
     }}
-    title={preview ? reactionType : undefined}
+    title={preview ? newKind : undefined}
     disabled={disabled}
   >
     <IconWithLabel
       icon={icon}
       count={count}
-      label={(preview || isMobile) ? undefined : reactionType}
+      label={(preview || isMobile) ? undefined : newKind}
     />
   </TxButton>
 }
 
 type VoterButtonsProps = VoterProps & {
-  only?: ReactionEnum,
+  kind?: ReactionEnum,
 }
 
-export const VoterButtons = (props: VoterButtonsProps) => {
-  const { post, only, ...voterProps } = props
-
-  const reaction = useAppSelector(state => selectPostReactionByPostId(state, post.id))
-
-  const renderVoterButton = useCallback((reactionType: ReactionEnum) => <VoterButton
-    reaction={reaction}
-    reactionEnum={reactionType}
-    post={post}
-    {...voterProps}
-  />, [ post.id, post.upvotesCount, post.downvotesCount, reaction?.kind || 'None' ])
-
-  const UpvoteButton = () => only !== ReactionEnum.Upvote ? renderVoterButton(ReactionEnum.Upvote) : null
-  const DownvoteButton = () => only !== ReactionEnum.Downvote ? renderVoterButton(ReactionEnum.Downvote) : null
+type InnerVoterButtonsProps = VoterButtonsProps & {
+  reaction?: ReactionStruct
+}
 
+const InnerVoterButtons = ({ kind, ...buttonProps }: InnerVoterButtonsProps) => {
   return <>
-    <UpvoteButton />
-    <DownvoteButton />
+    {kind !== ReactionEnum.Upvote &&
```

Comment:
If we delete `UpvoteVoterButton` & `DownvoteVoterButton`, then we can delete this comparisons `kind !== ... &&`